handler/schedule_test.go

```diff
@@ -90,72 +84,149 @@ func TestScheduleHandler_mergeResourcesWithDefaults(t *testing.T) {
+		"Given_GlobalDefaults_And_ScheduleDefaults_When_Spec_Then_UseSpec": {
 			globalCPUResourceLimit: "10m",
-			template: v1.ResourceRequirements{
+			givenScheduleTemplate: v1.ResourceRequirements{
 				Limits: v1.ResourceList{
 					v1.ResourceCPU: resource.MustParse("100m"),
 				},
 			},
-			resources: v1.ResourceRequirements{
+			givenResourceTemplate: v1.ResourceRequirements{
 				Limits: v1.ResourceList{
 					v1.ResourceCPU: resource.MustParse("200m"),
 				},
 			},
-			expected: v1.ResourceRequirements{
+			expectedTemplate: v1.ResourceRequirements{
 				Limits: v1.ResourceList{
 					v1.ResourceCPU: resource.MustParse("200m"),
 				},
 			},
 		},
 	}
 	cfg.Config = cfg.NewDefaultConfig()
-	for _, tt := range tests {
-		t.Run(tt.name, func(t *testing.T) {
+	for name, tt := range tests {
+		t.Run(name, func(t *testing.T) {
 			cfg.Config.GlobalCPUResourceLimit = tt.globalCPUResourceLimit
 			cfg.Config.GlobalCPUResourceRequest = tt.globalCPUResourceRequest
 			cfg.Config.GlobalMemoryResourceLimit = tt.globalMemoryResourceLimit
 			cfg.Config.GlobalMemoryResourceRequest = tt.globalMemoryResourceRequest
 			require.NoError(t, cfg.Config.ValidateSyntax())
-			s := ScheduleHandler{schedule: &v1alpha1.Schedule{Spec: v1alpha1.ScheduleSpec{
-				ResourceRequirementsTemplate: tt.template,
+			schedule := ScheduleHandler{schedule: &v1alpha1.Schedule{Spec: v1alpha1.ScheduleSpec{
+				ResourceRequirementsTemplate: tt.givenScheduleTemplate,
+			}}}
+			res := &v1alpha1.RunnableSpec{
+				Resources: tt.givenResourceTemplate,
+			}
+			schedule.mergeResourcesWithDefaults(res)
+			assert.Equal(t, tt.expectedTemplate, res.Resources)
+		})
+	}
+}
+
+func TestScheduleHandler_mergeBackendWithDefaults(t *testing.T) {
+	tests := map[string]struct {
+		globalS3Bucket       string
+		givenScheduleBackend v1alpha1.Backend
+		givenResourceBackend v1alpha1.Backend
+		expectedBackend      v1alpha1.Backend
+	}{
+		"Given_NoGlobalDefaults_And_NoScheduleDefaults_When_Spec_Then_UseSpec": {
+			givenResourceBackend: newS3Backend("https://resource-url", "resource-bucket"),
+			expectedBackend:      newS3Backend("https://resource-url", "resource-bucket"),
+		},
+		"Given_NoGlobalDefaults_And_ScheduleDefaults_When_NoSpec_Then_ApplyScheduleDefaults": {
+			givenScheduleBackend: newS3Backend("https://schedule-url", "schedule-bucket"),
+			expectedBackend:      newS3Backend("https://schedule-url", "schedule-bucket"),
+		},
+		"Given_NoGlobalDefaults_And_ScheduleDefaults_When_Spec_Then_UseSpec": {
+			givenScheduleBackend: newS3Backend("https://schedule-url", "schedule-bucket"),
+			givenResourceBackend: newS3Backend("https://resource-url", "resource-bucket"),
+			expectedBackend:      newS3Backend("https://resource-url", "resource-bucket"),
+		},
+		"Given_GlobalDefaults_And_NoScheduleDefaults_When_NoSpec_Then_UseGlobalDefaults": {
+			globalS3Bucket: "global-bucket",
+			givenScheduleBackend: v1alpha1.Backend{
```

Comment:
Is the `newS3Backend` function not usable here? en empty string "" is equal to not specifying in the struct, provided the property is not a pointer

Reply:
It certainly is, but my regex to did not catch this case 😅